remote/remote_test.go

```diff
@@ -42,75 +40,184 @@ func TestRemote(t *testing.T) {
+							ggcrv1.Platform{OS: "windows", Architecture: "arm"}),
+					)
+					h.AssertNil(t, err)
+					h.AssertNil(t, img.Save())
+
+					os, err := img.OS()
+					h.AssertNil(t, err)
+					h.AssertEq(t, os, "windows")
+
+					arch, err := img.Architecture()
+					h.AssertNil(t, err)
+					h.AssertEq(t, arch, "arm")
+				})
 			})
 		})
 
 		when("#FromRemoteBaseImage", func() {
 			when("base image exists", func() {
-				var (
-					baseName         = "busybox"
-					err              error
-					existingLayerSha string
-				)
+				it("sets the initial state from a linux/arm base image", func() {
+					baseImageName := "arm64v8/busybox@sha256:50edf1d080946c6a76989d1c3b0e753b62f7d9b5f5e66e88bef23ebbd1e9709c"
+					existingLayerSha := "sha256:5a0b973aa300cd2650869fd76d8546b361fcd6dfc77bd37b9d4f082cca9874e4"
 
-				it.Before(func() {
-					err = h.PullImage(dockerClient, baseName)
+					img, err := remote.NewImage(
+						repoName,
+						ggcrauthn.DefaultKeychain,
+						remote.FromBaseImage(baseImageName),
+					)
+					h.AssertNil(t, err)
+
+					os, err := img.OS()
 					h.AssertNil(t, err)
+					h.AssertEq(t, os, "linux")
 
-					inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), baseName)
+					arch, err := img.Architecture()
 					h.AssertNil(t, err)
+					h.AssertEq(t, arch, "arm64")
 
-					existingLayerSha = inspect.RootFS.Layers[0]
+					readCloser, err := img.GetLayer(existingLayerSha)
+					h.AssertNil(t, err)
+					defer readCloser.Close()
 				})
 
-				it("sets the initial state to match the base image", func() {
+				it("sets the initial state from a windows/amd64 base image", func() {
+					baseImageName := "mcr.microsoft.com/windows/nanoserver@sha256:06281772b6a561411d4b338820d94ab1028fdeb076c85350bbc01e80c4bfa2b4"
+					existingLayerSha := "sha256:26fd2d9d4c64a4f965bbc77939a454a31b607470f430b5d69fc21ded301fa55e"
+
 					img, err := remote.NewImage(
 						repoName,
-						authn.DefaultKeychain,
-						remote.FromBaseImage(baseName),
+						ggcrauthn.DefaultKeychain,
+						remote.FromBaseImage(baseImageName),
 					)
 					h.AssertNil(t, err)
 
+					os, err := img.OS()
+					h.AssertNil(t, err)
+					h.AssertEq(t, os, "windows")
+
+					arch, err := img.Architecture()
+					h.AssertNil(t, err)
+					h.AssertEq(t, arch, "amd64")
+
 					readCloser, err := img.GetLayer(existingLayerSha)
 					h.AssertNil(t, err)
 					defer readCloser.Close()
 				})
+
+				when("base image is a multi-OS/Arch manifest list", func() {
+					it("returns a base image matching the runtime GOOS/GOARCH", func() {
+						manifestListName := "golang:1.13.8"
+						existingLayerSha := "sha256:427da4a135b0869c1a274ba38e23d45bdbda93134c4ad99c8900cb0cfe9f0c9e"
+
+						img, err := remote.NewImage(
+							repoName,
+							ggcrauthn.DefaultKeychain,
+							remote.FromBaseImage(manifestListName),
+						)
+						h.AssertNil(t, err)
+
+						os, err := img.OS()
+						h.AssertNil(t, err)
+						h.AssertEq(t, os, "linux")
```

Comment:
Yeah, this is where it gets a little wacky. For a minimal constructor call: (say `local.NewImage(`/remote.NewImage(repoName, ggcrauthn.DefaultKeychain`)The `GOOS` would never used, and `remote` *always* defaults to Linux whereas `local` defaults to the Daemon OS.

Reply:
Yeah, this is where it get more subtle than expected. We discovered that `pack` almost always calls `imgutil` with a base image that already has the desired OS/Arch, so all we really need is some valid defaults that will be overridden in most cases. So we can get by without using `GOOS` at all, and have `remote` *always* default to Linux for empty images and have `local` default to the Daemon OS: 

```golang
remote.NewImage("test:tag", authn.DefaultKeychain) // always "linux/amd64"
local.NewImage("test:tag", dockerClient)   // image os/arch == docker os/arch
```

For a `imgutil` consumer (`pack`) to get a `windows` remote image, it would use a base image option when initializing:

```golang
remote.NewImage("test:tag", authn.DefaultKeychain, remote.FromBaseImage("nanoserver:1809"))
local.NewImage("test:tag", dockerClient, local.FromBaseImage("nanoserver:1809"))
```

I was going to add a new option to minimal new remote images (like:  `remote.WithDefaultPlatform("windows", "amd64")`) to get a `windows` image without using a `FromBaseImage` image, but it turns out `pack` wouldn't use it and it just added more complexity around supporting multiple options.

So the only reason either `remote_test.go` needs to know the daemon OS is for a couple tests that create containers for test assertions. If we refactored those tests, `remote_test.go` would be identical for any OS and we wouldn't event need a Daemon to run them (only a registry).

Reply:
Ohhh I see - thank you for the explanation! It might be worth updating the test description [here](https://github.com/buildpacks/imgutil/pull/42/files/d6c8f1ab604eb0b1e8a3075b6c6dcb88f6ed6cbd#diff-f1669c622581f9fccfe59ebda3685db3R162) to capture a bit of the nuance (assuming it doesn't get clobbered in the refactor).

I noticed you said "almost always" about `pack` calling `imgutil` with a base image - are there any edge cases that we need to be worried about?

Reply:
Yeah, the "almost always" for `pack` was for buildpackages which we haven't dug into yet. Presumably those would be "base image-less" images but we're not certain what exactly their needs are yet. There are potentially few different workarounds we could think of that could be performed outside of `imgutil` (by adding a "shim" base image or potentially always storing them as `linux` for remote) but we won't know until we dig in. It still may necessitate an `imgutil` change if those fall short.

Reply:
This makes sense, thank you for clarifying!